pt/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,333 @@
+  "map.flatten" => fn -> list |> Enum.map(map_fun) |> List.flatten() end
+})
+```
+
+Agora para executar nosso _benchmark_, chamamos:
+
+```shell
+$ mix run benchmark.exs
+```
+
+E devemos ver algo com a seguinte saída no seu console:
+
+```shell 
+Operating System: macOS
+CPU Information: Intel(R) Core(TM) i5-4260U CPU @ 1.40GHz
+Number of Available Cores: 4
+Available memory: 8.589934592 GB
+Elixir 1.5.1
+Erlang 20.0
+Benchmark suite executing with the following configuration:
+warmup: 2.00 s
+time: 5.00 s
+parallel: 1
+inputs: none specified
+Estimated total run time: 14.00 s
+
+
+Benchmarking flat_map...
+Benchmarking map.flatten...
+
+Name                  ips        average  deviation         median
+flat_map           1.03 K        0.97 ms    ±33.00%        0.85 ms
+map.flatten        0.56 K        1.80 ms    ±31.26%        1.60 ms
+
+Comparison:
+flat_map           1.03 K
+map.flatten        0.56 K - 1.85x slower
+```
+
+É claro que as informações e os resultados do seu sistema podem ser diferentes dependendo das especificações da máquina em que você está executando seus _benchmarks_, mas esta informação geral deve estar toda lá.
+
+À primeira vista, a seção `Comparison` nos mostra que a versão do nosso `map.flatten` é 1.85x mais lenta do que `flat_map` - muito útil saber! Mas vamos olhar as outras estatísticas que obtivemos:
+
+* **ips** - isso significa "iterações por segundo", que nos diz com que frequência a função pode ser executada em um segundo. Para esta métrica, um número maior é melhor.
+* **average** - este é o tempo médio de execução da função. Para esta métrica, um número baixo é melhor.
+* **deviation** - este é o desvio padrão, que nos diz o quanto os resultados para cada iteração variam nos resultados. Aqui é dado como uma porcentagem da média.
+* **median** - quando todos tempos medidos são ordenados, este é o valor médio (ou média dos dois valores do meio quando o número de amostras é par). Devido à inconsistências de ambiente este será mais estável do que a `average`, e um pouco mais provável que reflita a performance normal do seu código em produção. Para esta métrica, um número baixo é melhor.
+
+Há também outras estatísticas disponíveis, mas estas quatro são frequentemente as mais úteis e comumente usadas para _benchmarking_,  por isso elas são exibidas no formatador padrão. Para aprender mais sobre outras métricas disponíveis, confira a documentação [hexdocs](https://hexdocs.pm/benchee/Benchee.Statistics.html#statistics/1).
+
+# Configuração
+
+Uma das melhores partes do Benchee são todas as opções de configuração disponíveis. Examinaremos o básico primeiro, uma vez que não requerem exemplos de código, e então mostraremos como usar uma das melhores características do Benchee - _inputs_.
+
+## Básico
+
+Benchee possui uma grande variedade de opções de configuração. Na interface mais comum `Benchee.run/2`, estes são passados como segundo argumento na forma de uma _keywork list_ opcional:
```

Comment:
I believe "(...) est**a**s são passad**a**s como segundo argumento (...)" is more correct, since we are referencing the options (feminine noun in Portuguese).